### PR TITLE
Replace point and vector particle emitter with spheroid emitter

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Replace emitVelocity and velocitySpread with orientation, speed, and speedSpread.
 Make the emitter a spheroid with particles emitted perpendicular to its surface.
 Make able to emit from part of its volume and surface.
 Update existing scripts that use particles.

### CODE REVIEW

**Comment by ZappoMan** [comment]
If I'm reading this correctly, you appear to be adding some new properties and abandoning some older properties. If that's the case, then it would be PREFERABLE to reuse the enum slots for the properties you are abandoning for some of your new properties. In general we'd like to keep the total number of properties down. It's also possible to alias some of the previously used property slots for other entity types so we keep this list shorter.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 03194437f5f2fa4a5eadfe7fc36a2a66f144c700 available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

**Comment by ctrlaltdavid** [comment]
@ZappoMan I've reused the relevant enum slots, I think without breaking things.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ec146069b14e97e86f80de360935bb6d483f5d82 available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 60ba6e174fa675a4b09e929d020e2658dc987f84 available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

**Comment by cozza13** [comment]
I have QA'ed this

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 1]
```diff
@@ -258,20 +288,26 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
         READ_ENTITY_PROPERTY(PROP_TEXTURES, QString, setTextures);
-        READ_ENTITY_PROPERTY(PROP_VELOCITY_SPREAD, glm::vec3, setVelocitySpread);
+        if (args.bitstreamVersion < VERSION_ENTITIES_PARTICLE_ELLIPSOID_EMITTER) {
+            // OLD PROP_VELOCITY_SPREAD FAKEOUT
+            SKIP_ENTITY_PROPERTY(PROP_SPEED_SPREAD, glm::vec3);
+        }
     } else {
-        // EMIT_STRENGTH FAKEOUT
+        // EMIT_ACCELERATION FAKEOUT
         READ_ENTITY_PROPERTY(PROP_PARTICLE_RADIUS, float, setParticleRadius);
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: why don't you change this to use the new SKIP_PROPERTY() macro?

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 2]
```diff
@@ -258,20 +288,26 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-        // LOCAL_GRAVITY FAKEOUT
+        // ACCELERATION_SPREAD FAKEOUT
         READ_ENTITY_PROPERTY(PROP_PARTICLE_RADIUS, float, setParticleRadius);
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: why don't you change this to use the new SKIP_PROPERTY() macro?

**Comment by murillodigital** [comment]
test this please

File: libraries/entities/src/ParticleEffectEntityItem.cpp (hunk 2 above)
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: actually this old code is confusing... does it make sense to you? can you explain it?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 60ba6e174fa675a4b09e929d020e2658dc987f84 available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

File: libraries/entities/src/ParticleEffectEntityItem.cpp (hunk 2 above)
Comment (ctrlaltdavid) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: @eric Thoughts about what to do here? E.g., comment explaining exactly what's going on and/or using SKIP_PROPERTY macro?

Comment (ericrius1) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: I would defer to @ZappoMan's judgment on this one.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2b902238e73d85142eed36b8b04d666ab025da11 available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2b902238e73d85142eed36b8b04d666ab025da11 available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

**Comment by ZappoMan** [comment]
@ctrlaltdavid - please repair these warnings... we want all PRs to be warning free... Soon jenkins will consider all warnings as build failures.

```
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp: In member function ‘void ParticleEffectEntityItem::stepSimulation(float)’:
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:707:54: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     azimuth = _azimuthStart + (2.0 * PI + _azimuthFinish - _azimuthStart) * randFloat();
                                                      ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:707:59: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     azimuth = _azimuthStart + (2.0 * PI + _azimuthFinish - _azimuthStart) * randFloat();
                                                           ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:707:76: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     azimuth = _azimuthStart + (2.0 * PI + _azimuthFinish - _azimuthStart) * randFloat();
                                                                            ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:707:103: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     azimuth = _azimuthStart + (2.0 * PI + _azimuthFinish - _azimuthStart) * randFloat();
                                                                                                       ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:707:103: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:724:75: warning: implicit conversion from ‘float’ to ‘__gnu_cxx::__promote_2<float, int, float, double>::__type {aka double}’ to match other operand of binary expression [-Wdouble-promotion]
                         radiusScale = 1.0f - std::pow(1.0f - randRadius, 3);
                                                                           ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:728:57: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     float x = radiuses.x * cos(elevation) * cos(azimuth);
                                                         ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:729:57: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     float y = radiuses.y * cos(elevation) * sin(azimuth);
                                                         ^
/data/jenkins/Jenkins/workspace/hifi-pull-request-ubuntu/libraries/entities/src/ParticleEffectEntityItem.cpp:730:57: warning: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
                     float z = radiuses.z * sin(elevation);
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3d44fdfdf5fc56bd563129d46437b91587ed36df available at: https://gist.github.com/67b05d0f7ab86bb29287<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5843.dmg

**Comment by ctrlaltdavid** [comment]
@ZappoMan That appears to have fixed those warnings.

**Pull request merged**